src/custom-element.js

```diff
@@ -117,96 +117,113 @@ export function stubElements(win) {
 
 
 /**
+ * Determines the layout of the element from the layout attribute, and if
+ * not specified, by considering hasNaturalDimensions() and the width,
+ * height, and sizes attributes.
+ * @param {!AmpElement} element
+ * @param {string} widthAttr
+ * @param {string} heightAttr
+ * @return {!Layout}
+ */
+function determineLayout(element, widthAttr, heightAttr) {
+  const layoutAttr = element.getAttribute('layout');
+  if (layoutAttr) {
+    return parseLayout(layoutAttr);
+  }
+  if (hasNaturalDimensions(element.tagName)) {
```

Comment:
It still not entirely correct. Plus the code is now duplicated with the code below. Maybe I should get a pass on simplifying this?